tools/src/main/java/io/narayana/db/DBAllocator.java

```diff
@@ -0,0 +1,328 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.db;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+/**
+ * @author <a href="mailto:karm@redhat.com">Michal Karm Babacek</a>
+ */
+public class DBAllocator extends Allocator {
+    private static final Logger LOGGER = Logger.getLogger(DBAllocator.class.getName());
+
+    DBAllocator() {
+        // Use getInstance
+    }
+
+    public DB allocateDB(final int expiryMinutes) {
+        // Something between nothing and a business week...
+        if (expiryMinutes < 1 || expiryMinutes > 7200) {
+            throw new IllegalArgumentException("expiryMinutes out of range 1 - 7200");
+        }
+
+        final String projectBuildDirectory = System.getProperty("project.build.directory");
+        if (StringUtils.isEmpty(projectBuildDirectory)) {
+            throw new IllegalArgumentException("project.build.directory must not be empty");
+        }
+        final File driversDir = new File(projectBuildDirectory);
+        if (!driversDir.exists()) {
+            throw new IllegalArgumentException(driversDir.getAbsolutePath() + " must exist");
+        }
+
+        // Timeout for fetching DB driver
+        if (StringUtils.isEmpty(System.getProperty("dballocator.driver.url.timeout.seconds"))) {
+            throw new IllegalArgumentException("dballocator.driver.url.timeout.seconds must not be empty");
+        }
+        final long driverFetchTimeoutSeconds = Long.parseLong(System.getProperty("dballocator.driver.url.timeout.seconds"));
+        if (driverFetchTimeoutSeconds < 0 || driverFetchTimeoutSeconds > 1800) {
+            throw new IllegalArgumentException("driverFetchTimeoutSeconds out of range 0 - 1800");
+        }
+
+        // Fetching driver
+        final String dballocatorDriverUrl = System.getProperty("dballocator.driver.url");
+        if (StringUtils.isEmpty(dballocatorDriverUrl)) {
+            throw new IllegalArgumentException("dballocator.driver.url must not be empty");
+        }
+        final URL driverJarSrc;
+        try {
+            driverJarSrc = new URL(dballocatorDriverUrl);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+            LOGGER.severe(e.getMessage());
+            return null;
+        }
+        final String[] driverJarFilename = dballocatorDriverUrl.split("/");
```

Comment:
please, use  https://docs.oracle.com/javase/7/docs/api/java/io/File.html#getName(). In case the handling is needed do not use the `"/"` but https://docs.oracle.com/javase/7/docs/api/java/io/File.html#separator